benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -330,10 +332,11 @@ private void handleRunningTestState(String testID) throws BenchFlowTestIDDoesNot
         exit = true;
       }
       // Exit while waiting for the Experiment Manager to notify about the scheduled
-      // experiment to be executed.
+      // experiment to be executed or aborted (terminating state)
       else if (prevTestRunningState != null && testRunningState != null
```

Comment:
and also add a check that `RunninStatesHandler.terminating(...)` was called, for instance here https://github.com/benchflow/benchflow/blob/a19d162c400ca964b60158f6144b6baa901e2003/benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/scheduler/TestTaskSchedulerIT.java#L378